### ISSUE TITLE
Callable: Not nullable, inconsistent behaviour

### ISSUE THREAD

**Issue by Structed** [godotengine/godot#72209]
### Godot version

v4.0.beta16.official [518b9e580]

### System information

Windows 11

### Issue description

The new Callable has a few issues/inconsistencies around nullability:

1. Callables can be assigned `null`, but it fails at runtime.
  > `Trying to assign value of type 'Nil' to a variable of type 'Callable'`
  Static analysis does not currently find this issue.
2. Callables have a method `is_null()`, but right now, they cannot be `null`.
  a. What is the purpose of this method, if Callables cannot be `null`?
  b. As an alternative, there is also `is_valid()`
3. Function declarations with an optional `Callable` parameter need to define them by instantiating a `Callable()` with it's default constructor

# How should it work?
Either:
* Callables could/should be `null`able
or
* Static analysis should not allow assignment of `null` to a `Callable`
  And should potentially hint at using an instance with the default constructor.

## Failing Scenario
1. Create a function which accepts an optional parameter of type `Callable`, by setting it's default to `null`
  ```gdscript
func call_me_not_working_optional_param(callback: Callable = null):
	var test: Callable	
	test = null	
	if callback != null && callback.is_valid():
		callback.call()
		print("called")
	else:
		print("not called")
  ```
2. Call `call_me_not_working_optional_param()`
3. Start the build in debug
4. When The code is executed, an error is thrown:
  > `Trying to assign value of type 'Nil' to a variable of type 'Callable'`

### Minimal reproduction project

[godot4-callable-null.zip](https://github.com/godotengine/godot/files/10523453/godot4-callable-null.zip)

**Comment by dalexeev** [comment]
1. `Callable`, as well as other basic Variant types like `Array` and `Color`, are not nullable. You must use `Callable()` if you want to pass an empty `Callable`.
2. Assigning `null` must be a compile time error. This is a bug that will be fixed by PR #71844.
3. The name `is_null()` is not very good, but it's just the opposite of `is_valid()`. I think this naming comes from C++. We have `Ref<T>` with `is_null()` and `is_valid()` methods.

**Comment by Structed** [comment]
Thanks, that's what I thought, that it should not be billable. But didn't find the other bug. Thanks for the reference.

Apart from that: I assume the `is_null()` will not be considered to be changed?
